js/modules/k6/ws/ws.go

```diff
@@ -334,13 +335,27 @@ func (s *Socket) handleEvent(event string, args ...goja.Value) {
 	}
 }
 
-func (s *Socket) Send(message string) {
-	// NOTE: No binary message support for the time being since goja doesn't
-	// support typed arrays.
+// Send writes the given string or ArrayBuffer message to the connection.
+func (s *Socket) Send(message interface{}) {
 	rt := common.GetRuntime(s.ctx)
 
-	writeData := []byte(message)
-	if err := s.conn.WriteMessage(websocket.TextMessage, writeData); err != nil {
+	var (
+		msgType int
+		msg     []byte
+	)
+	switch m := message.(type) {
+	case string:
+		msgType = websocket.TextMessage
+		msg = []byte(m)
+	case goja.ArrayBuffer:
+		msgType = websocket.BinaryMessage
+		msg = m.Bytes()
+	default:
+		err := fmt.Errorf("unsupported message type: %T, expected string or ArrayBuffer", message)
+		common.Throw(common.GetRuntime(s.ctx), err)
+	}
+
+	if err := s.conn.WriteMessage(msgType, msg); err != nil {
```

Comment:
I disagree that it's cleaner, but here you go: b80813c3.

Expanding the API to now include another method that does the same thing for a different data type is hardly a good idea. It might've worked fine for that user and it's an isolated solution, but it's another method we'll have to eventually deprecate.

@na-- WDYT?